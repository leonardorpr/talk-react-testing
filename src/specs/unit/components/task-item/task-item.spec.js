import React from 'react'
import { render, fireEvent } from 'react-testing-library'
import 'jest-styled-components'

import TaskItem from 'components/task-item'

describe('[COMPONENTS] TaskItem', () => {
  it('Should render correctly', () => {
    const { container } = render(<TaskItem time='15 minutes' label='Make Coffee' done toggle={jest.fn()} />)
    expect(container).toMatchSnapshot()
  })

  it('Should check if a task is not completed', () => {
    const { container } = render(<TaskItem time='15 minutes' label='Make Coffee' done={false} toggle={jest.fn()} />)
    expect(container.querySelector('#indicator')).toHaveStyleRule('background-color', '#DA493A')
  })

  it('Should check if a task is completed', () => {
    const { container } = render(<TaskItem time='15 minutes' label='Make Coffee' done toggle={jest.fn()} />)
    expect(container.querySelector('#indicator')).toHaveStyleRule('background-color', '#2ECC71')
  })

  it('Should complete the task when it was clicked 2 times', () => {
    const { container } = render(<TaskItem time='15 minutes' label='Make Coffee' done={false} toggle={jest.fn()} />)
    fireEvent.doubleClick(container.querySelector('#card'))
    expect(container.querySelector('#indicator')).toHaveStyleRule('background-color', '#2ECC71')
  })
})
