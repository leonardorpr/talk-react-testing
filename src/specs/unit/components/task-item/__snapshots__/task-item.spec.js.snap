// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENTS] TaskItem Should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  max-width: 400px;
  height: 60px;
  margin-left: 20px;
  cursor: pointer;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
  margin: 15px;
}

.c0:hover {
  -webkit-box-shadow: 0px 2px 17px -5px rgba(0,0,0,0.75);
  -moz-box-shadow: 0px 2px 17px -5px rgba(0,0,0,0.75);
  box-shadow: 0px 2px 17px -5px rgba(0,0,0,0.75);
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
}

.c2 {
  background-color: #2A2C30;
  width: 97.5%;
  border-top-right-radius: 6px;
  border-bottom-right-radius: 6px;
  padding: 10px;
}

.c1 {
  background-color: #2ECC71;
  width: 2.5%;
  margin-right: 2px;
  border-top-left-radius: 6px;
  border-bottom-left-radius: 6px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  text-align: center;
  text-transform: uppercase;
  font-weight: bold;
  color: #FFF;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c5 {
  text-align: end;
  font-size: 12px;
  color: #FFF;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          Fazer caf√©
        </span>
        <span
          class="c5"
        >
          15 minutos
        </span>
      </div>
    </div>
  </div>
</div>
`;
